internal/auth/authn_repo_test.go

```diff
@@ -19,20 +19,51 @@ func TestRepositoryAuthenticator(t *testing.T) {
 
 	require.NoError(t, repo.WriteSession(ctx, env.Repository, repo.WriteSessionOptions{},
 		func(ctx context.Context, w repo.RepositoryWriter) error {
-			p := &user.Profile{
-				Username: "user1@host1",
+			for _, tc := range []struct {
+				profile  *user.Profile
+				password string
+			}{
+				{
+					profile: &user.Profile{
+						Username:            "user1@host1",
+						PasswordHashVersion: crypto.HashVersion1,
+					},
+					password: "password1",
+				},
+				{
+					profile: &user.Profile{
+						Username:               "user2@host2",
+						KeyDerivationAlgorithm: crypto.ScryptAlgorithm,
+					},
+					password: "password2",
+				},
+				{
+					profile: &user.Profile{
+						Username: "user3@host3",
+					},
+					password: "password3",
+				},
+			} {
+				tc.profile.SetPassword(tc.password)
+				err := user.SetUserProfile(ctx, w, tc.profile)
+				if err != nil {
+					return err
+				}
 			}
-
-			p.SetPassword("password1", crypto.DefaultKeyDerivationAlgorithm)
-
-			return user.SetUserProfile(ctx, w, p)
+			return nil
 		}))
 
 	verifyRepoAuthenticator(ctx, t, a, env.Repository, "user1@host1", "password1", true)
 	verifyRepoAuthenticator(ctx, t, a, env.Repository, "user1@host1", "password2", false)
 	verifyRepoAuthenticator(ctx, t, a, env.Repository, "user1@host1", "password11", false)
 	verifyRepoAuthenticator(ctx, t, a, env.Repository, "user1@host1a", "password1", false)
 	verifyRepoAuthenticator(ctx, t, a, env.Repository, "user1@host1a", "password1a", false)
+
+	// Test for password with KeyDerivationSet
+	verifyRepoAuthenticator(ctx, t, a, env.Repository, "user2@host2", "password2", true)
+
+	// Test for User with neither key derivation or PasswordHashVersion set
+	verifyRepoAuthenticator(ctx, t, a, env.Repository, "user3@host3", "password3", false)
```

Comment:
Don't we want a `true` here instead of `false` ? The default algorithm should be scrypt or has that changed ?

Reply:
nvm, this is the case where neither `PasswordHashVersion` and nor `KeyDerivationAlgorithm` are set

this makes sense now